common/src/main/java/org/opensearch/ml/common/conversational/ConversationalIndexConstants.java

```diff
@@ -0,0 +1,103 @@
+    public final static String META_ENDED_FIELD = "last_interaction_time";
+    /** Name of the metadata field for number of interactions */
+    public final static String META_LENGTH_FIELD = "num_interactions";
+    /** Name of the metadata field for name of the conversation */
+    public final static String META_NAME_FIELD = "name";
+    /** Name of the owning user field in all indices */
+    public final static String USER_FIELD = "user";
+    /** Mappings for the conversational metadata index */
+    public final static String META_MAPPING = "{\n"
+        + "    \"properties\": {\n"
+        + "        \""
+        + META_NAME_FIELD
+        + "\": {\"type\": \"keyword\"},\n"
+        + "        \""
+        + META_CREATED_FIELD
+        + "\": {\"type\": \"date\", \"format\": \"strict_date_optional_time||epoch_millis\"},\n"
+        + "        \""
+        + META_ENDED_FIELD
+        + "\": {\"type\": \"date\", \"format\": \"strict_date_optional_time||epoch_millis\"},\n"
+        + "        \""
+        + META_LENGTH_FIELD
+        + "\": {\"type\": \"integer\"},\n"
+        + "        \""
+        + USER_FIELD
+        + "\": {\"type\": \"keyword\"}\n"
+        + "    }\n"
+        + "}";
+
+        /** Name of the conversational interactions index */
+        public final static String INTERACTIONS_INDEX_NAME = ".conversational-interactions";
+        /** Name of the interaction field for the conversation Id */
+        public final static String INTERACTIONS_CONVERSATION_ID_FIELD = "conversation_id";
+        /** Name of the interaction field for the human input */
+        public final static String INTERACTIONS_INPUT_FIELD = "input";
+        /** Name of the interaction field for the prompt template */
+        public final static String INTERACTIONS_PROMPT_FIELD = "prompt";
+        /** Name of the interaction field for the AI response */
+        public final static String INTERACTIONS_RESPONSE_FIELD = "response";
+        /** Name of the interaction field for the GAI Agent */
+        public final static String INTERACTIONS_AGENT_FIELD = "agent";
+        /** Name of the interaction field for the timestamp */
+        public final static String INTERACTIONS_TIMESTAMP_FIELD = "timestamp";
+        /** Name of the interaction field for any excess metadata */
+        public final static String INTERACTIONS_METADATA_FIELD = "metadata";
+        /** Mappings for the interactions index */
+        public final static String INTERACTIONS_MAPPINGS = "{\n"
+        + "    \"properties\": {\n"
+        + "        \""
+        + INTERACTIONS_CONVERSATION_ID_FIELD
+        + "\": {\"type\": \"keyword\"},\n"
+        + "        \""
+        + INTERACTIONS_TIMESTAMP_FIELD
+        + "\": {\"type\": \"date\", \"format\": \"strict_date_optional_time||epoch_millis\"},\n"
+        + "        \""
+        + INTERACTIONS_INPUT_FIELD
+        + "\": {\"type\": \"text\"},\n"
+        + "        \""
+        + INTERACTIONS_PROMPT_FIELD
```

Comment:
Why do we need to store prompt? I assume the prompt rarely change in a conversation and could be huge. So it is not meaningful to store too many redundant prompts in the interactions index which is space wasteful too.

Reply:
We're hoping that long term there will be a prompting framework and this can just refer to an id. But also, we want as much explainability as we can possibly get, since LLMs are magic - so we want to know (and we think users want to be able to find out) _exactly_ what got sent to the LLM and what the LLM said back

Reply:
* maybe remove / move to convoMeta